pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -503,6 +535,26 @@ static void update_conn_stats(
 
     // If already in our map, increment size in-place
     if (val != NULL) {
+        // If the type is TCP and it's the first time we see this conn (sent = 0, recv = 0)
+        // assign it an ID
+        if (type == CONN_TYPE_TCP && val->sent_bytes == 0 && val->recv_bytes == 0) {
+            // Have to unset the PID to access the tcp_stats map
+            t.pid = 0;
+
+            tcp_stats_t* tcp_val;
+            // initialize-if-no-exist
+            tcp_stats_t empty = {};
+            bpf_map_update_elem(&tcp_stats, &t, &empty, BPF_NOEXIST);
+            tcp_val = bpf_map_lookup_elem(&tcp_stats, &t);
+            if (tcp_val != NULL && tcp_val->id == 0) {
+                // Use fetch and add for safety, since this gets executed only when id == 0 it gives us what we want
```

Comment:
I think you want to do a compare and swap with zero. This will do the wrong thing if we handle the same conn_tuple_t on different processors at the same time

Reply:
👍

Reply:
It seems we can't use `__sync_val_compare_and_swap` helpers in eBPF (they do not compile :/) see [here](https://gist.github.com/brendangregg/731cf2ce54bf1f9a19d4ccd397625ad9#file-cpuunclaimed-py-L208).

I will check if it's possible for a connection's packets to be processed by multiple different CPUs or if it's always the same one

Reply:
I just checked and this is possible 😞 , I will see if there can be an other way of setting the id in a safe way